src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -8553,5 +8543,48 @@ public void ParamsArray_ThisModifier_02()
                 // var lam = (params this int[] xs) => xs.Length;
                 Diagnostic(ErrorCode.ERR_ThisInBadContext, "this").WithLocation(1, 19));
         }
+
+        [Fact]
+        public void ImplicitlyTypedLambdaWithModifier_CSharp13()
+        {
+            var source = """
+                delegate void D(ref int i);
+
+                class C
+                {
+                    void M()
+                    {
+                        D d = (ref a) => { };
+                    }
+                }
+                """;
+            CreateCompilation(source, parseOptions: TestOptions.Regular13).VerifyDiagnostics(
+                // (7,16): error CS8652: The feature 'simple lambda parameter modifiers' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         D d = (ref a) => { };
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "ref a").WithArguments("simple lambda parameter modifiers").WithLocation(7, 16));
+        }
+
+        [Fact]
+        public void IncorrectlyTypedLambdaWithCorrectRefNess()
+        {
+            var source = """
+                delegate void D(ref int i);
+
+                class C
+                {
+                    void M()
+                    {
+                        D d = (ref byte a) => { };
+                    }
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (7,25): error CS1678: Parameter 1 is declared as type 'ref byte' but should be 'ref int'
+                //         D d = (ref byte a) => { };
+                Diagnostic(ErrorCode.ERR_BadParamType, "a").WithArguments("1", "ref ", "byte", "ref ", "int").WithLocation(7, 25),
+                // (7,28): error CS1661: Cannot convert lambda expression to type 'D' because the parameter types do not match the delegate parameter types
+                //         D d = (ref byte a) => { };
+                Diagnostic(ErrorCode.ERR_CantConvAnonMethParams, "=>").WithArguments("lambda expression", "D").WithLocation(7, 28));
+        }
```

Comment:
Looks like it should be since we added params support in c# 12: https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/lambda-method-group-defaults.md

This is a challenging part of spec'ing as i based my changes off of the spec in csharplang, but it's hard to tell what additional changes have happened on top of htat.  i'll revise.

Reply:
Added tests for this scenario.